File: Fuentes/Notificaciones Correo/Entidades/EmpresasPrincipal.vb
Imports System.Data.SqlServerCe

Public Class EmpresasPrincipal

    Private idEmpresa As Integer
    Private activa As Boolean
    Private instanciaSql As String
    Private rutaBd As String
    Private usuarioSql As String
    Private contrasenaSql As String

    Public Property EIdEmpresa() As Integer
        Get
            Return Me.idEmpresa
        End Get
        Set(value As Integer)
            Me.idEmpresa = value
        End Set
    End Property
    Public Property EActiva() As Boolean
        Get
            Return Me.activa
        End Get
        Set(value As Boolean)
            Me.activa = value
        End Set
    End Property
    Public Property EInstanciaSql() As String
        Get
            Return Me.instanciaSql
        End Get
        Set(value As String)
            Me.instanciaSql = value
        End Set
    End Property
    Public Property ERutaBd() As String
        Get
            Return rutaBd
        End Get
        Set(value As String)
            rutaBd = value
        End Set
    End Property
    Public Property EUsuarioSql() As String
        Get
            Return usuarioSql
        End Get
        Set(value As String)
            usuarioSql = value
        End Set
    End Property
    Public Property EContrasenaSql() As String
        Get
            Return contrasenaSql
        End Get
        Set(value As String)
            contrasenaSql = value
        End Set
    End Property

    Public Function ObtenerPredeterminada() As List(Of EmpresasPrincipal)

        Dim lista As New List(Of EmpresasPrincipal)()
        Try
            Dim comando As New SqlCeCommand()
            comando.Connection = BaseDatos.conexionPrincipal
            comando.CommandText = "SELECT * FROM Empresas WHERE Activa='TRUE'"
            BaseDatos.conexionPrincipal.Open()
            Dim dataReader As SqlCeDataReader = Nothing
            dataReader = comando.ExecuteReader()
            Dim empresasPrincipal As EmpresasPrincipal
            While (dataReader.Read())
                empresasPrincipal = New EmpresasPrincipal()
                empresasPrincipal.EIdEmpresa = Convert.ToInt32(dataReader("IdEmpresa"))
                empresasPrincipal.EActiva = Convert.ToBoolean(dataReader("Activa").ToString())
                empresasPrincipal.EInstanciaSql = dataReader("InstanciaSql").ToString()
                empresasPrincipal.ERutaBd = dataReader("RutaBd").ToString()
                empresasPrincipal.EUsuarioSql = dataReader("UsuarioSql").ToString()
                empresasPrincipal.EContrasenaSql = dataReader("ContrasenaSql").ToString()
                lista.Add(empresasPrincipal)
            End While
            BaseDatos.conexionPrincipal.Close()
            Return lista
        Catch ex As Exception
            Throw ex
        End Try

    End Function

End Class

File: Fuentes/Principal/Entidades/EmpresasPrincipal.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Data.SqlServerCe;
using System.Linq;
using System.Text;

namespace Entidades
{
    public class EmpresasPrincipal
    {

        private int idEmpresa;
        private bool activa;
        private string instanciaSql;
        private string rutaBd;
        private string usuarioSql; 
        private string contrasenaSql;

        public int IdEmpresa
        {
            get { return this.idEmpresa; }
            set { this.idEmpresa = value; }
        }
        public bool Activa
        {
            get { return this.activa; }
            set { this.activa = value; }
        }
        public string InstanciaSql
        {
            get { return this.instanciaSql; }
            set { this.instanciaSql = value; }
        }
        public string RutaBd
        {
            get { return rutaBd; }
            set { rutaBd = value; }
        }
        public string UsuarioSql
        {
            get { return usuarioSql; }
            set { usuarioSql = value; }
        }
        public string ContrasenaSql
        {
            get { return contrasenaSql; }
            set { contrasenaSql = value; }
        }

        public List<EmpresasPrincipal> ObtenerPredeterminada()
        {

            List<EmpresasPrincipal> lista = new List<EmpresasPrincipal>();
            bool conexionCorrecta = false;
            while (!conexionCorrecta)
            { 
                try
                {
                    SqlCeCommand comando = new SqlCeCommand();
                    comando.Connection = BaseDatos.conexionPrincipal;
                    comando.CommandText = "SELECT * FROM Empresas WHERE Activa='TRUE'";
                    BaseDatos.conexionPrincipal.Open(); 
                    SqlCeDataReader dataReader = default(SqlCeDataReader);
                    dataReader = comando.ExecuteReader();
                    EmpresasPrincipal empresasPrincipal;
                    while ((dataReader.Read()))
                    {
                        empresasPrincipal = new EmpresasPrincipal();
                        empresasPrincipal.IdEmpresa = Convert.ToInt32(dataReader["IdEmpresa"]);
                        empresasPrincipal.Activa = Convert.ToBoolean(dataReader["Activa"].ToString());
                        empresasPrincipal.InstanciaSql = dataReader["InstanciaSql"].ToString();
                        empresasPrincipal.RutaBd = dataReader["RutaBd"].ToString();
                        empresasPrincipal.UsuarioSql = dataReader["UsuarioSql"].ToString();
                        empresasPrincipal.ContrasenaSql = dataReader["ContrasenaSql"].ToString();
                        lista.Add(empresasPrincipal);
                    }
                    BaseDatos.conexionPrincipal.Close();
                    conexionCorrecta = true;
                    //return lista;
                }
                catch (SqlCeException ex)
                {
                    if (ex.NativeError == 25035)
                    {
                        conexionCorrecta = false;
                    }
                    else
                    {
                        conexionCorrecta = true;
                        throw ex;
                    }
                }
                catch(Exception ex)
                {
                    conexionCorrecta = true;
                    throw ex; 
                }
                finally
                {
                    BaseDatos.conexionPrincipal.Close();
                }
            }
            return lista;

        }

    }
}
